test/e2e/agentgateway_e2e_test.go

```diff
@@ -105,8 +89,10 @@ var _ = Describe("Agent Gateway", Ordered, func() {
 				"metadata": map[string]interface{}{},
 			},
 		}
-		body, err := utils.PostRequest(gatewayUrl+"/mocked-agent-exposed-1", payload)
+		body, statusCode, err := utils.MakeGatewayPost("default", "agent-gateway", 10000,
+			"/mocked-agent-exposed-1", payload)
 		Expect(err).NotTo(HaveOccurred(), "Failed to send POST request to agent gateway")
+		Expect(statusCode).To(Equal(200), "Expected HTTP 200 status code")
```

Comment:
You could also wrap the request and expects into one Eventually. This way, we retry failed requests, i.e. if the agent is not yet ready.

This way, we could simplify the setup, as we do not have to wait for gateway and agent to be ready.